src/main/java/org/usfirst/frc4048/common/swervev2/SwerveDrivetrain.java

```diff
@@ -1,170 +1,102 @@
-        PID drivePid = PID.of(Constants.DRIVE_PID_P,Constants.DRIVE_PID_I,Constants.DRIVE_PID_D);
-        PID steerPid = PID.of(Constants.STEER_PID_P,Constants.STEER_PID_I,Constants.STEER_PID_D);
-        Gain driveGain = Gain.of(Constants.DRIVE_PID_FF_V,Constants.DRIVE_PID_FF_S);
-        Gain steerGain = Gain.of(Constants.STEER_PID_FF_V,Constants.STEER_PID_FF_S);
-        frontLeft = new SparkMaxSwerveModule(encodedSwerveSparkMaxFL, drivePid,steerPid, driveGain,steerGain,constraints);
-        frontRight = new SparkMaxSwerveModule(encodedSwerveSparkMaxFR, drivePid,steerPid, driveGain,steerGain,constraints);
-        backLeft = new SparkMaxSwerveModule(encodedSwerveSparkMaxBL, drivePid,steerPid, driveGain,steerGain,constraints);
-        backRight = new SparkMaxSwerveModule(encodedSwerveSparkMaxBR, drivePid,steerPid, driveGain,steerGain,constraints);
-        frontLeftDrive.setInverted(true);
-        frontRightDrive.setInverted(false);
-        backRightDrive.setInverted(false);
-        backLeftDrive.setInverted(true);
-        navxGyro.setAngleAdjustment(gyroOffset);
+    public SwerveDrivetrain(SwerveIdConfig frontLeftConfig, SwerveIdConfig frontRightConfig, SwerveIdConfig backLeftConfig, SwerveIdConfig backRightConfig,
+                            KinematicsConversionConfig conversionConfig, SwervePidConfig pidConfig, Gyro gyro) 
+    {
+        this.gyro = gyro;
+        EncodedSwerveSparkMax encodedSwerveSparkMaxFL = new EncodedSwerveMotorBuilder(frontLeftConfig, conversionConfig).build();
+        EncodedSwerveSparkMax encodedSwerveSparkMaxFR = new EncodedSwerveMotorBuilder(frontRightConfig, conversionConfig).build();
+        EncodedSwerveSparkMax encodedSwerveSparkMaxBL = new EncodedSwerveMotorBuilder(backLeftConfig, conversionConfig).build();
+        EncodedSwerveSparkMax encodedSwerveSparkMaxBR = new EncodedSwerveMotorBuilder(backRightConfig, conversionConfig).build();
+
+        this.frontLeft = new GenericSwerveModule(encodedSwerveSparkMaxFL, pidConfig.getDrivePid(),pidConfig.getSteerPid(),pidConfig.getDriveGain(),pidConfig.getSteerGain(),pidConfig.getGoalConstraint());
+        this.frontRight = new GenericSwerveModule(encodedSwerveSparkMaxFR, pidConfig.getDrivePid(),pidConfig.getSteerPid(),pidConfig.getDriveGain(),pidConfig.getSteerGain(),pidConfig.getGoalConstraint());
+        this.backLeft = new GenericSwerveModule(encodedSwerveSparkMaxBL, pidConfig.getDrivePid(),pidConfig.getSteerPid(),pidConfig.getDriveGain(),pidConfig.getSteerGain(),pidConfig.getGoalConstraint());
+        this.backRight = new GenericSwerveModule(encodedSwerveSparkMaxBR, pidConfig.getDrivePid(),pidConfig.getSteerPid(),pidConfig.getDriveGain(),pidConfig.getSteerGain(),pidConfig.getGoalConstraint());
+        this.frontRight.getSwerveMotor().getDriveMotor().setInverted(true);
+        this.frontLeft.getSwerveMotor().getDriveMotor().setInverted(false);
```

Comment:
These seem inverted from before. Is that intentional?

Reply:
I just move the lines by accident it should have the same functionality